### PR TITLE
Install udev rules on generic linux target

### PR DESCRIPTION
**Pull request by rbekken**

### CODE REVIEW

**Comment by paul-jewell** [comment]
Can we change the name of the 45-openpilot-permissions.rules to be 45-librepilot-permissions.rules?

**Comment by rbekken** [comment]
If its to be renamed in the repository, I could simply rename it to udev.rules?

**Comment by parched** [comment]
Hi rbekken, welcome and thank you for your contribution. This is something we have been meaning to do. We would like this to be off by default as some packaging (i.e., debian handles this separately from the install rule).

**Comment by parched** [comment]
Yes I agree the file should be renamed too. udev.rules seems fine

**Comment by parched** [comment]
or maybe something generic but relevant like 45-uav.rules

File: package/Linux.mk [hunk 1]
```diff
@@ -105,5 +107,4 @@ install:
 	$(V1) sed -i -e 's/gcs/$(GCS_SMALL_NAME)/g;s/GCS/$(GCS_BIG_NAME)/g;s/org/$(ORG_SMALL_NAME)/g;s/ORG/$(ORG_BIG_NAME)/g' $(DESTDIR)$(datadir)/applications/$(ORG_SMALL_NAME).desktop
 
 	$(V1) $(INSTALL) -T $(ROOT_DIR)/package/linux/openpilot.png $(DESTDIR)$(datadir)/pixmaps/$(ORG_SMALL_NAME).png
-
-
+	$(V1) install -c -m 644 $(ROOT_DIR)/package/linux/45-openpilot-permissions.rules $(DESTDIR)$(sysconfdir)/udev/rules.d/45-$(ORG_SMALL_NAME).rules
```
Comment (parched) [inline package/Linux.mk]: Also it would be better to follow the convention here with $(INSTALL) unless there is a reason to break it.

Comment (rbekken) [inline package/Linux.mk]: $(INSTALL) would leave the files with 664 rater than 644. I guess that could be corrected in the repo thought.

**Comment by parched** [comment]
Yeah just change it in the repo, it should be 644

**Comment by parched** [comment]
Out of interest what distro are you on?

**Comment by rbekken** [comment]
Fedora

**Comment by theothercliff** [comment]
About udev permissions rules.  Doesn't it refer to the manufacturer of the hardware, so OpenPilot is correct in 45-openpilot-permissions.rules

**Comment by rbekken** [comment]
Cancel pull request as this is broken in so many ways...

**Comment by parched** [comment]
@theothercliff What happens if we add other non-OP hardware? If we left it as openpilot when installed it would conflict anyway

**Comment by theothercliff** [comment]
If the user runs OP code and also runs LP code, they will get two exact sets of the same rules for e.g. Revo.
If we add support for other brand hardware, we just add a new file.  The exact file that the hardware manufacturer uses.  This seems the best way, but I am not an expert.

**Comment by parched** [comment]
I guess we could do it that way. But they would still need to have different names like
45-librepilot-gcs-openpilot.rules
45-librepilot-gcs-other_hardware.rules

Normally a udev file will be per package not per hardware vendor though.

**Comment by parched** [comment]
@rbekken I think to option to enable the udev rules install should be named `enable-udev-rules` which you can have set to `no` by default then check if it isn't `no` in which install

**Comment by rbekken** [comment]
Seems like a good idea. Im not all that familiar with Makefiles scripting but I'll look into it. Just have to do more of testing and less committing :P

**Comment by parched** [comment]
What's your email address @rbekken ?

**Comment by rbekken** [comment]
i sent you a mail

**Pull request closed without merging**